src/arrayMethodJoin.js

```diff
@@ -5,7 +5,24 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let newSeparator = separator;
+
+    if (separator === undefined) {
+      newSeparator = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
```

Comment:
try implicit conditional:
- `null` and `undefined` are equivalent to false when used in conditional blocks
- so `this[i]` would replace your current conditional
   - and `!this[i]` would mean `this[i] === null || this[i] === undefined`

try it